docs/index.md

```diff
@@ -116,51 +116,60 @@ $ npm test
 
 ## Run Cycle Overview
 
-A brief outline on the order Mocha's components are executed.
-Worth noting that all hooks, `describe` and `it` callbacks are run in the order they are defined (i.e. found in the file).
-
-```
-run 'mocha spec.js'
-|
-spawn child process
-|
-|--------------> inside child process
-  process and apply options
-  |
-  run spec file/s
-  |
-  |--------------> per spec file
-    suite callbacks (e.g., 'describe')
-    |
-    'before' root-level pre-hook
-    |
-    'before' pre-hook
-    |
-    |--------------> per test
-      'beforeEach' root-level pre-hook
-      |
-      'beforeEach' pre-hook
-      |
-      test callbacks (e.g., 'it')
-      |
-      'afterEach' post-hook
-      |
-      'afterEach' root-level post-hook
-    |<-------------- per test end
-    |
-    'after' post-hook
-    |
-    'after' root-level post-hooks
-  |<-------------- per spec file end
-|<-------------- inside child process end
-```
+> Updated for v9.0.0.
+
+The following is a mid-level outline of Mocha's "flow of execution" when run in Node.js; the "less important" details have been omitted.
+
+In a browser, test files are loaded by `<script>` tags, and calling `mocha.run()` begins at step 9 [below](#serial-mode).
+
+### Serial Mode
```

Comment:
Worth splitting into parsing and execution phases?

Reply:
I'm not convinced I can make it any clearer by drawing a line there. do you have any specific suggestions on how to explain it clearly?

Reply:
Not for clarity, just to codify our definition of where these phases start and what's included. Was just imagining a simple divide I.e.
```
--- Parsing phase
1. ....
...
4. ...
---
--- Execution phase
5. ....
----
`
But they are mainly useful for internal purposes anyway so happy to drop. Leave with you.

Reply:
hmmm......


I'm not sure _I_ know what the definition is.  certainly, we run our suite callbacks (I think this is what you're referring to with "parsing phase") in their entirety before our hook and test callbacks.  In the changes I've made, step 8. refers to this:

> 8. The (default) [bdd interface](#bdd) loads the test files _in no particular order_, which are given an interface-specific `global` context (this is how, e.g., `describe()` ends up as a global in a test file)
>     1. When a test file is loaded, Mocha executes all of its suites and finds--_but does not execute_--any hooks and tests therein.
>     2. Top-level hooks, tests and suites are all made members of an "invisible" _root suite_; there is only _one_ root suite for the entire process

after that, we run global setup fixtures, which really has nothing to do with the suites.  after _that_, we run the hook/test callbacks.  so to me anyway, there's no clear "logical grouping" of tasks, like you've written above.  it's just a sequence of steps.

now... we could make an argument that global setup fixtures should actually run _before_ test files are loaded.  that _might_ make more sense, but it would mean that we'd have to expose APIs for programmatic users to trigger setup _and_ teardown execution manually.  maybe someone wants this, but a) you can toggle the behavior on and off anyway via the public `Mocha#enableGlobalSetup()` and `Mocha#enableGlobalTeardown()` methods, and b) it's not very friendly to make programmatic users go thru extra hoops to get standard behavior.  

however, this brings up the point that `Mocha#runGlobalSetup()` and `Mocha#runGlobalTeardown()` should be public, if a programmatic users wants to fiddle with when these run:

```js
const mocha = new Mocha();
mocha.addFile('/some/file.js');
mocha.enableGlobalSetup(false);
// run the global setup early, before mocha.run() is called
await mocha.runGlobalSetup();
// do something else here
mocha.run();
```

Reply:
Hmm interesting.
Sorry I should have said, my understanding is the parsing phase is just options and validation. So it's very short. I think all the loading of files and running comes under execution phase. But that's just my own mental model.